apps/web/src/actions/auth.ts

```diff
@@ -39,6 +41,58 @@ export async function createAccount(data: z.infer<typeof signupSchema>) {
     }
 }
 
+export async function forgotPassword(data: z.infer<typeof forgotSchema>) {
+    const { mail } = data;
+
+    const user = await getUserByMail(mail);
+    if (!user) {
+        throw new Error("E-Mail wird nicht verwendet.");
+    }
+
+    const uuid = randomBytes(20).toString("hex");
+
+    await createToken({tokenId: uuid, userId: user.id, create: new Date() });
+
+    const resetUrl = `${process.env.NEXTAUTH_URL}/reset?token=${uuid}`;
+    const html = `Diese E-Mail wurde als Antwort auf Ihre Anfrage gesendet, Ihr Passwort zurückzusetzen. Bitte klicken Sie hierzu auf den Link unten. Aus Sicherheitsgründen ist dieser nur eine Stunde gültig.
+<br> <a href="${resetUrl}">${resetUrl}</a>
+<br> Wenn Sie dies nicht angefordert haben, empfehlen wir Ihnen, Ihre Passwörter zu ändern.`;
+
+    await SendMail(mail, "Energyleaf: Passwort zurückseten", html)
+}
+
+export async function resetPassword(data: z.infer<typeof resetSchema>, token_id: string | null) {
+    const { password, passwordRepeat } = data;
+
+    const token = await getToken(token_id);
+    if (token === null) {
+        // relativ unpräzise Error Message ist ein feature, um möglichst wenig Angriffainformation zu bieten
```

Comment:
Man könnte das sogar noch unpräziser machen :D 
"Das Zurücksetzen des Passworts ist nicht möglich" oder so

-> Aber ich würde empfehlen, dass wir den String unten in ne Konstante auslagern, da wir ja die gleich Msg mehrfach werfen oder macht man das nicht im Web-Kontext?

Reply:
Wird selten gemacht.
Meist ein eigener Error Type wenn es sich häufig wiederholt.

Wir können auch ein Object zurückgeben direkt mit der Message die wir anzeigen wollen. Das wäre für i18n eventuell gut.

```typescript
return {
    error: "Ungültiges oder abgelaufenes Passwort-Reset-Token"
};
```

Läuft am ende alles auf das gleich hinaus. Wichtig ist, wie wir die Nachricht dem Nutzer im Frontend anzeigen und wie wir uns entscheiden, dass im Code zu behandeln.